### PR TITLE
table columns text length can be set in constructor

### PR DESCRIPTION
**Pull request by Nestorferrando**

### CODE REVIEW

File: kuick-repositories-squash/src/jvmMain/kotlin/kuick/repositories/squash/ModelRepositorySquash.kt [hunk 1]
```diff
@@ -35,16 +36,15 @@ open class ModelRepositorySquash<I : Any, T : Any>(
...
-                    returnType == String::class.starProjectedType -> varchar(columnName, LONG_TEXT_LEN)
+                    returnType == String::class.starProjectedType -> varchar(columnName, textLength)
                     returnType == Int::class.starProjectedType -> integer(columnName)
                     returnType == Long::class.starProjectedType -> long(columnName)
                     returnType == Double::class.starProjectedType -> decimal(columnName, 5, 4)
                     returnType == Boolean::class.starProjectedType -> bool(columnName)
                     returnType == Date::class.starProjectedType -> long(columnName)
                     returnType == LocalDate::class.starProjectedType -> varchar(columnName, LOCAL_DATE_TIME_LEN)
                     returnType.isSubtypeOf(Id::class.starProjectedType) -> (varchar(columnName, ID_LEN))
-                    returnType.toString().startsWith("koan") -> varchar(columnName, LONG_TEXT_LEN)
```
Comment (soywiz) [inline kuick-repositories-squash/src/jvmMain/kotlin/kuick/repositories/squash/ModelRepositorySquash.kt]: @maballesteros is this used?

File: kuick-repositories-squash/src/jvmMain/kotlin/kuick/repositories/squash/ModelRepositorySquash.kt [hunk 2]
```diff
@@ -35,16 +36,15 @@ open class ModelRepositorySquash<I : Any, T : Any>(
             //println("Registering field ${prop} with return type: ${prop.returnType}")
             with(table) {
                 var columnDefinition: ColumnDefinition<Any?> = when {
-                    returnType == String::class.starProjectedType -> varchar(columnName, LONG_TEXT_LEN)
+                    returnType == String::class.starProjectedType -> varchar(columnName, textLength)
```
Comment (soywiz) [inline kuick-repositories-squash/src/jvmMain/kotlin/kuick/repositories/squash/ModelRepositorySquash.kt]: Also won't it be common to want to specify the max length per column instead for the whole table?
What about keeping this change so you can provide a default for the table, but also being able to specify it in the ColumnDefinition itself?

Comment (Nestorferrando) [reply kuick-repositories-squash/src/jvmMain/kotlin/kuick/repositories/squash/ModelRepositorySquash.kt]: I agree, being able to optionally define column length can be quite useful. Each string can store different stuff and thus have different requirements

File: kuick-repositories-squash/src/jvmMain/kotlin/kuick/repositories/squash/ModelRepositorySquash.kt (hunk 1 above)
Comment (maballesteros) [reply kuick-repositories-squash/src/jvmMain/kotlin/kuick/repositories/squash/ModelRepositorySquash.kt]: This was used to handle `koan` value object types... but I think it's already covered by the `else` clause.

Comment (soywiz) [reply kuick-repositories-squash/src/jvmMain/kotlin/kuick/repositories/squash/ModelRepositorySquash.kt]: I missed that part. Yeah, the else has the same code so it should work :)

**Pull request merged**